.github/workflows/pr_check.yml

```
@@ -0,0 +1,67 @@
+name: PR Test
+on:
+  push:
+    branches:
+      - 'master'
+      - 'develop'
+      - 'releases/*'
+  pull_request:
+    types:
+      - opened
+      - synchronize
+      - reopened
+      - ready_for_review
+
+# This is what will cancel the workflow concurrency
+concurrency:
+  group: ${{ github.workflow }}-${{ github.event.pull_request.number || github.ref }}
+  cancel-in-progress: true
+
+defaults:
+  run:
+    shell: bash
+jobs:
+  pr_test:
+    if: |
+      github.event.pull_request.draft == false &&
+      !(startsWith(github.event.pull_request.title, '[WIP]')) &&
+      !(startsWith(github.event.pull_request.title, '[Dependent]'))
+    strategy:
+      fail-fast: false
+      matrix:
+        os: ['ubuntu-20.04', 'windows-2022']
+        python-version: ['3.10']
```

```suggestion
        python-version: ['3.9']
```
If we're removing checks for all the matrix, it makes sense to leave the oldest supported version checks in PRs,